packages/flutter_rearch/example/lib/models/todo.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter_mimir/flutter_mimir.dart';
+
+/// Represents an item in the todos list.
+class Todo implements Comparable<Todo> {
+  /// Creates a new [Todo].
+  const Todo({
+    required this.timestamp,
+    required this.title,
+    required this.completed,
+    this.description,
+  });
+
+  /// The timestamp of the todo in milliseconds since epoch.
+  ///
+  /// Used as the primary key for the todo.
+  final int timestamp;
+
+  /// The title of the todo.
+  final String title;
+
+  /// The description of the todo.
+  final String? description;
+
+  /// Whether the todo is completed.
+  final bool completed;
+
+  /// Toggles the completion status of the todo.
+  // Should have a @useResult annotation. Check if it is ok to import
+  // package:meta
+  Todo toggleCompletion() => _copyWith(completed: !completed);
+
+  @override
+  int compareTo(Todo other) {
+    return timestamp.compareTo(other.timestamp);
+  }
+
+  Todo _copyWith({
+    int? timestamp,
+    String? title,
+    String? Function()? description,
+    bool? completed,
+  }) {
+    return Todo(
+      timestamp: timestamp ?? this.timestamp,
+      title: title ?? this.title,
+      description: description != null ? description() : this.description,
```

Comment:
really wish dart had a better way to handle this... I'd normally say let's pull in `freezed` but frankly that's a little much for just this